src/OrchardCore.Modules/OrchardCore.Autoroute/Drivers/AutoroutePartDisplay.cs

```diff
@@ -117,7 +115,7 @@ public override async Task<IDisplayResult> UpdateAsync(AutoroutePart model, IUpd
                 // This can only validate the path if the Autoroute is not managing content item routes or the path is absolute.
                 if (!String.IsNullOrEmpty(model.Path) && (!settings.ManageContainedItemRoutes || (settings.ManageContainedItemRoutes && model.Absolute)))
                 {
-                    var possibleConflicts = await _session.QueryIndex<AutoroutePartIndex>(o => o.Path == model.Path).ListAsync();
+                    var possibleConflicts = await _session.QueryIndex<AutoroutePartIndex>(o => (o.Published || o.Latest) && o.Path == model.Path).ListAsync();
```

Comment:
@jtkech as far as I know the index only ever contains published or latest. so do we need the extra check here?

Reply:
@deanmarcussen 

Yes we need it as we now also have removed items (only when soft removed, not all versions)

So with Published = Latest = false

Reply:
Ah ok, so soft removed is allowing for if a document that is not latest, or published, to also be removed?

In which case the table now has Published = false and Latest = false

Sorry i'm being dumb here, but I just want to check what will end up in the table (permanently)

Does that mean it then stays in the table permanently, as an item that has been removed?
So the index is now tracking more document history? (like the `ContentItemIndex` does, but not items that have been superceded - so not removed, but replaced by the another draft)

Reply:
Yes, but the `AutoroutePartIndex` table will have much less record than `ContentItemIndex`, not all versions e.g. when you saved a draft and / or each time you publish an item, there is no new record.

For a given item, it will have just one more record and only if you soft remove it

Reply:
> Sorry i'm being dumb here

No problem ;)

Reply:
> Ah ok, so soft removed is allowing for if a document that is not latest, or published, to also be removed?

Sorry, when i say soft remove, it is as we already do, set published and latest to false, soft because the version is still in the database.

So this is only at the time we removed an item (we know that through the handler part) that it is still indexed.

E.g. when we publish a draft, the draft version is removed, but it is not indexed as we don't trigger the removed handler event

Reply:
So, for a given item, it will have just one more record

And only if you remove it, and only if it has an autoroute part